catshadow/client_docker_test.go

```diff
@@ -66,7 +66,7 @@ func createCatshadowClientWithState(t *testing.T, stateFile string, useReunion b
 	cfg, err := catshadowCfg.ClientConfig()
 	require.NoError(err)
 
-	cfg, linkKey, err := client.AutoRegisterRandomClient(cfg)
+	cfg, linkKey, err := client.NewEphemeralClient(cfg)
```

Comment:
we could also have called it NewClient but maybe it's helpful to have multiple constructors so that we continue to support the old use case aka loopix.